File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.go
func missingMultiple(nums []int, k int) int {
    m := map[int]bool{}
    for _, num := range nums {
        m[num] = true
    }

    num := k
    for num < 1000 {
        if !m[num] {
            return num
        }

        num += k
    }

    return num
}

// array, hash map
// time: O(n + t)
// space: O(n)

File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.py
class Solution:
    def missingMultiple(self, nums: List[int], k: int) -> int:
        s = set()
        for num in nums:
            s.add(num)
        
        num = k
        while num < 1000:
            if num not in s:
                return num
            num += k
        
        return num

# array, hash map
# time: O(n + t)
# space: O(n)